fdbclient/ManagementAPI.actor.cpp

```diff
@@ -2886,24 +2886,11 @@ ACTOR Future<std::vector<BulkLoadTaskState>> getBulkLoadTasksWithinRange(Databas
 }
 
 ACTOR Future<Void> setBulkLoadSubmissionTransaction(Transaction* tr, BulkLoadTaskState bulkLoadTask) {
+	ASSERT(normalKeys.contains(bulkLoadTask.getRange()) &&
+	       (bulkLoadTask.phase == BulkLoadPhase::Submitted ||
+	        (bulkLoadTask.phase == BulkLoadPhase::Complete && bulkLoadTask.hasEmptyData())));
 	tr->setOption(FDBTransactionOptions::LOCK_AWARE);
 	tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
-	if (bulkLoadTask.phase != BulkLoadPhase::Submitted) {
-		TraceEvent(g_network->isSimulated() ? SevError : SevWarnAlways, "SubmitBulkLoadTaskError")
-		    .setMaxEventLength(-1)
-		    .setMaxFieldLength(-1)
-		    .detail("Reason", "WrongPhase")
-		    .detail("Task", bulkLoadTask.toString());
-		throw bulkload_task_failed();
-	}
-	if (!normalKeys.contains(bulkLoadTask.getRange())) {
-		TraceEvent(g_network->isSimulated() ? SevError : SevWarnAlways, "SubmitBulkLoadTaskError")
-		    .setMaxEventLength(-1)
-		    .setMaxFieldLength(-1)
-		    .detail("Reason", "RangeOutOfScope")
-		    .detail("Task", bulkLoadTask.toString());
-		throw bulkload_task_failed();
-	}
```

Comment:
Why drop these checks?

Reply:
Oh, I will add those back in the next PR.

Reply:
I change these soft assert to the hard assert at the beginning of the function.